### PR TITLE
Fix #14 - unused methods

### LINKED ISSUE #14: Unused methods
[Two methods (related to exception handling)](https://github.com/twingly/twingly-search-api-python/blob/afc0a502567ec0dc592dfffac79eab92a85f30c6/twingly_search/parser.py#L74-L79) in the Parser class aren't used. That also means that the function `TwinglyException.from_api_response_message` is not used either.

Instead the errors are [handled directly in the `.parse` method](https://github.com/twingly/twingly-search-api-python/blob/afc0a502567ec0dc592dfffac79eab92a85f30c6/twingly_search/parser.py#L33-L38).

In Ruby we do it like this: https://github.com/twingly/twingly-search-api-ruby/blob/72cec57ae11a1f865f9044d7bf1c7de38a2cb147/lib/twingly/search/parser.rb#L15-L18

### PR DESCRIPTION
**Pull request by bearburger**

### CODE REVIEW

**Comment by bearburger** [comment]
Can you please mention me if this is bug. Just sometimes missing messages :-)

File: twingly_search/parser.py [hunk 1]
```diff
@@ -27,15 +35,13 @@ def parse(self, document):
...
         if doc.find('{http://www.twingly.com}operationResult') is not None:
             if doc.find('{http://www.twingly.com}operationResult').attrib['resultType'] == 'failure':
-                if 'API key' in doc.find('{http://www.twingly.com}operationResult').text:
-                    raise TwinglyAuthException(doc.find('{http://www.twingly.com}operationResult').text)
-                else:
-                    raise TwinglyServerException(doc.find('{http://www.twingly.com}operationResult').text)
+                self._handle_failure(doc.find('{http://www.twingly.com}operationResult'))
+
+
+        if unicode('twinglydata') != unicode(doc.tag):
```
Comment (roback) [inline twingly_search/parser.py]: Why is `unicode()` needed for this?

**Comment by bearburger** [comment]
Because in Python **2** `doc.tag` returns `str` type and 'twinglydata' string has `unicode` type so comparison always returns false. So we should cast both of them to unicode. We could simply use `unicode` function without additional imports but there is no `unicode` function in Python **3** and it should be replaced with `str`.

**Comment by roback** [comment]
> Because in Python 2 doc.tag returns str type and 'twinglydata' string has unicode type. And comparison always returns false. So we should cast both of them to unicode. We could simply use unicode function without additional imports but there is no unicode function in Python 3 and it should be replaced with str.

I tested with python locally, and for me the comparison seems to work:

``` python
$ python
Python 2.7.11 (default, Dec  9 2015, 11:02:16)
[GCC 4.2.1 Compatible Apple LLVM 7.0.0 (clang-700.1.76)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> u'test' == 'test'
True
```

**Comment by walro** [comment]
> I tested with python locally, and for me the comparison seems to work:

Not if you add a unicode character to your comparison.

**Comment by roback** [comment]
> Not if you add a non-unicode character to your comparison.

True, but can the string be something other than "twinglydata" or "html"?

**Comment by bearburger** [comment]
@walro maybe @roback is right. Under Linux I haven't errors so probably this is Windows specific issue. And there is almost zero chance that you will get non-ASCII character as tag name.

**Comment by walro** [comment]
> True, but can the string be something other than "twinglydata" or "html"?

Nope :)

**Comment by roback** [comment]
LGTM

Thanks for the fix @bearburger :smile:

**Pull request merged**